### PR TITLE
Add graceful shutdown to pulpcore-worker

### LINKED ISSUE #3940: As a user I can gracefully shut down a pulpcore-worker
A user should be able to stop a `pulpcore-worker` in such a way that allows the worker's current task to run to completion.

### PR DESCRIPTION
**Pull request by mdellweg**
fixes #3940

### CODE REVIEW

File: pulpcore/tasking/pulpcore_worker.py [hunk 1]
```diff
@@ -407,6 +413,7 @@ def child_signal_handler(sig, frame):
     # If you kill the process a second time it's not graceful anymore.
     signal.signal(signal.SIGINT, signal.SIG_DFL)
     signal.signal(signal.SIGTERM, signal.SIG_DFL)
+    signal.signal(signal.SIGHUP, signal.SIG_DFL)
     signal.signal(signal.SIGUSR1, signal.SIG_DFL)
 
     if sig == signal.SIGUSR1:
```
Comment (lubosmj) [inline pulpcore/tasking/pulpcore_worker.py:419]: Does `sys.exit()` trigger any signal? If not, can we move this if-condition up?

Comment (mdellweg) [reply pulpcore/tasking/pulpcore_worker.py:419]: What for?
But now that you ask, maybe we don't want to restore the SIGHUP handler at all, so the graceful kill stays graceful...

Comment (lubosmj) [reply pulpcore/tasking/pulpcore_worker.py:419]: Setting the signal handlers back to `signal.SIG_DFL` seems to be useless if we will eventually exit the process when `sig == signal.SIGUSR1`.

Comment (mdellweg) [reply pulpcore/tasking/pulpcore_worker.py:419]: The idea is that after telling the process to quit (via signalling) and it did not quit in time, we want all the signals to behave normal when we kill it a second time (which should never ever be needed, but you know, software...).

File: pulpcore/tasking/pulpcore_worker.py [hunk 2]
```diff
@@ -114,14 +114,18 @@ def __init__(self):
...
     def _signal_handler(self, thesignal, frame):
-        # Reset signal handlers to default
-        # If you kill the process a second time it's not graceful anymore.
-        signal.signal(signal.SIGINT, signal.SIG_DFL)
-        signal.signal(signal.SIGTERM, signal.SIG_DFL)
-
-        _logger.info(_("Worker %s was requested to shut down."), self.name)
-
-        self.task_grace_timeout = TASK_GRACE_INTERVAL
+        if thesignal == signal.SIGHUP:
```
Comment (dkliban) [inline pulpcore/tasking/pulpcore_worker.py:117]: Kubernetes sends a SIGTERM. We should be handling that signal gracefully. 

https://cloud.google.com/blog/products/containers-kubernetes/kubernetes-best-practices-terminating-with-grace

**Review by dkliban — APPROVED** [review]

**Review by bmbouter — APPROVED** [review]
This looks good to me! This will be really helpful.

**Pull request merged**